File: Lifecycle/app/src/main/java/com/example/lifecycle/MainActivity.java
package com.example.lifecycle;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        android.widget.Toast.makeText(getApplicationContext(), "On Create", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Create");
    }

    @Override
    protected void onStart() {
        super.onStart();
        android.widget.Toast.makeText(getApplicationContext(), "On Start", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Start");
    }

    @Override
    protected void onResume() {
        super.onResume();
        android.widget.Toast.makeText(getApplicationContext(), "On Resume", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Resume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        android.widget.Toast.makeText(getApplicationContext(), "On Pause", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Pause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        android.widget.Toast.makeText(getApplicationContext(), "On Stop", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Stop");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        android.widget.Toast.makeText(getApplicationContext(), "On Restart", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Restart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        android.widget.Toast.makeText(getApplicationContext(), "On Destroy", Toast.LENGTH_SHORT).show();
        Log.d("lifecycle", "On Destroy");
    }
}

File: RXjavaproject/app/src/main/java/com/example/rxjavaproject/MainActivity.kt
package com.example.rxjavaproject

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import com.jakewharton.rxbinding4.view.clicks
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.disposables.Disposable
import io.reactivex.rxjava3.schedulers.Schedulers
import retrofit2.Retrofit
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.util.Observer

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val button = findViewById<Button>(R.id.btnClick)
        simpleObservable()
        button.clicks().subscribe{
            Log.d("nava", "Button clicked")
        }
        implementNetworkCall()
    }

    private fun simpleObservable() {
        val observable = listOf<String>("A", "B", "C")

    }

    private fun implementNetworkCall(){
        val retrofit = Retrofit.Builder()
            .baseUrl("https://fakestoreapi.com")
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
            .build()

        val ProductService = retrofit.create(ProductService::class.java)
        ProductService.getProducts()
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe{
                Log.d("nava", it.toString())
        }
    }
}
